### ISSUE TITLE
Other extension which interferes with new automatic saving of bookmark option.

### ISSUE THREAD

**Issue by user0022** [gildas-lormeau/SingleFile#361]
Hello! I was not sure of opening a new thread or continuing the first issue which was: [Feature Question - saving existing bookmarks](https://github.com/gildas-lormeau/SingleFile/issues/320)? Well I opened a new page.

I use the [MySessions](https://addons.mozilla.org/en-CA/firefox/addon/my-sessions/?src=search) extension, and there's a little problem with it, it is an extension which remembers navigation sessions, but well it saves windows and tabs in a "MySessions" folder placed in Bookmarks: " As a repository, browser bookmarks are selected as the most reliable and secure storage.".
And a new SingleFile/Z option, when enabled, automatically saves a page that one bookmarks. 
So when saving a session, SingleFile/Z begins to save all pages of all windows (I just stopped it with a click on SF/Z icon, but finally I disable the SF/Z option for the moment).
And this might happen with other extensions which would use bookmarks like MySessions! I've checked in addons by typing "bookmark", there are [597 extensions which say "bookmark"](https://addons.mozilla.org/en-CA/firefox/search/?q=bookmark&type=extension) for a reason, how many others cause the same problem? 0 or 20? 600 to check!! Or just wait for the complaints.
I hope it's repairable.

**Comment by gildas-lormeau** [comment]
Well, I do not really know what to do here. Technically speaking, it's impossible for SingleFile to know that a bookmark has been created by the user or an extension. So, SingleFile cannot ignore ones created by MySessions automatically. By the way, I don't think there are a lot of extensions that use bookmarks for a rather special purpose, like MySessions does.

The only thing that I could propose would be to give you the ability to blacklist some bookmark folders in SingleFile (e.g. the one used by MySessions).

**Comment by user0022** [comment]
Thank you, the blacklisting of a folder seems a good solution to my impression!
Even if if I understand, the weakness that makes you unsatisfied is that it is not general/cross-extension, but only for this one, or maybe one-by-one if other similar cases occur.
Using the MySessions extension AND this option could be rare, but possibly one day not-so-few other users would meet the same problem!

**
Anyway I'm gonna go on using the SF option and the MySessions addon for the moment, even without the possible new feature you described, I'll just disable option or addon, or switch profile, or else.
Otherwise what I might do (one day, but not sure).
-Ask the MySessions developper (he might have the same answer but who knows, he could give an option to place sessions somewhere else for example, if he finds it worth it).
-MySessions replacement by a similar addon.

**
Maybe also the exclusion of folders would have another good use, if a user would want to use the option "bookmark+save" but in same time use the traditionnal bookmark purpose (which is to remember an empty url), so he could do it in a maybe faster/easier way than switching profile or disabling option/addon when chosing the destination folder when the prompt/pop-up displays asking it before confirmation.

**Comment by gildas-lormeau** [comment]
I agree that it would be better to address this issue for the reasons you mentioned. I think I'll add a hidden option soon to allow the blacklisting of bookmark folders. Then I'll see if there's a need to make the option visible to users and easily editable or not.

**Comment by gildas-lormeau** [comment]
I implemented the hidden option `ignoredBookmarkFolders` in the last commit. I also updated the [wiki page](https://github.com/gildas-lormeau/SingleFile/wiki/Hidden-options). 

If you want to ignore the bookmark folder "MySessions" (I don't know the name used by MySessions), replace (in the exported settings file):
`ignoredBookmarkFolders: []`
with
`ignoredBookmarkFolders: ["MySessions"]`

The feature will be available in the next version I'll publish soon.
